src/main/scala/com/spotify/spark/bigquery/BigQueryClient.scala

```diff
@@ -109,8 +110,29 @@ private[bigquery] class BigQueryClient(conf: Configuration) {
    */
   def load(gcsPath: String, destinationTable: TableReference,
            writeDisposition: WriteDisposition.Value = null,
-           createDisposition: CreateDisposition.Value = null): Unit = {
+           createDisposition: CreateDisposition.Value = null,
+           isPartitionedByDay: Boolean = false): Unit = {
```

Comment:
Maybe instead of `isPartitionByDay` and `Boolean` make this `TimePartitioning` and default to `null` thus making it more flexible and add helpers for different partitions like `DayPartitioning` in the package level.